unit_tests/testStrings/testStrings.f90

```diff
@@ -8,83 +8,88 @@
 !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
 PROGRAM testStrings
 #include "UnitTest.h"
-  USE ISO_C_BINDING
-  USE UnitTest
-  USE IntrType
-  USE Strings
-
-  IMPLICIT NONE
-
-  CREATE_TEST('TEST STRINGS')
-  REGISTER_SUBTEST('Un-Iniitialized State',testUnInit)
-  REGISTER_SUBTEST('Character Assignments',testAssign_chars)
-  REGISTER_SUBTEST('Numberal Assignments',testAssign_nums)
-  REGISTER_SUBTEST('Array Assignments',testAssign_arrays)
-  REGISTER_SUBTEST('Intrinsics',testIntrinsic)
-  REGISTER_SUBTEST('string_functs',testStrFunct)
-  FINALIZE_TEST()
+USE ISO_C_BINDING
+USE UnitTest
+USE IntrType
+USE Strings
+
+IMPLICIT NONE
+
+CREATE_TEST('TEST STRINGS')
+REGISTER_SUBTEST('Un-Iniitialized State',testUnInit)
+REGISTER_SUBTEST('Character Assignments',testAssign_chars)
+REGISTER_SUBTEST('Numberal Assignments',testAssign_nums)
+REGISTER_SUBTEST('Array Assignments',testAssign_arrays)
+REGISTER_SUBTEST('Intrinsics',testIntrinsic)
+REGISTER_SUBTEST('string_functs',testStrFunct)
+FINALIZE_TEST()
 !
 !-------------------------------------------------------------------------------
 CONTAINS
 !
 !-------------------------------------------------------------------------------
 !>
-  SUBROUTINE testUnInit()
-    TYPE(StringType) :: testString
-    COMPONENT_TEST('Test Unitialized')
-    !Test methods for uninitialized state
-    ASSERT_EQ(LEN(testString),0,'LEN(testString) (uninit)')
-    ASSERT_EQ(LEN_TRIM(testString),0,'LEN_TRIM(testString) (uninit)')
-    ASSERT_EQ(CHAR(testString),'','CHAR(testString) (uninit)')
-    ASSERT_EQ(TRIM(testString),'','TRIM(testString) (uninit)')
-  ENDSUBROUTINE testUnInit
+SUBROUTINE testUnInit()
+  TYPE(StringType) :: testString
+  COMPONENT_TEST('Test Unitialized')
+  !Test methods for uninitialized state
+  ASSERT_EQ(LEN(testString),0,'LEN(testString) (uninit)')
+  ASSERT_EQ(LEN_TRIM(testString),0,'LEN_TRIM(testString) (uninit)')
+  ASSERT_EQ(CHAR(testString),'','CHAR(testString) (uninit)')
+  ASSERT_EQ(TRIM(testString),'','TRIM(testString) (uninit)')
+
+  COMPONENT_TEST('Test Clear')
+  CALL testString%clear()
+  ASSERT_EQ(LEN(testString),0,'LEN(testString) (uninit)')
+  ASSERT_EQ(LEN_TRIM(testString),0,'LEN_TRIM(testString) (uninit)')
+  ASSERT_EQ(CHAR(testString),'','CHAR(testString) (uninit)')
+  ASSERT_EQ(TRIM(testString),'','TRIM(testString) (uninit)')
```

Comment:
Should actually assign something to the string and then test clear as well

Reply:
So I added something for this, but the order is weird since the assignments have been checked yet.